### PR TITLE
fix some issue for sycl and hip build

### PR DESCRIPTION
**Pull request by xyuan**
some minor changes to fix sycl and hip build issues

### CODE REVIEW

**Review by mrnorman — APPROVED** [review]
This looks fine to me. I'm alarmed at how long that `this->is_initialized()` but sat there in the code :-/. It should have segfaulted the code from day one on Nvidia and AMD! Thanks for finding that and fixing it.

**Comment by mrnorman** [comment]
By the way, you said this runs with SYCL. Did you find a way around the `_wp` FP literal suffix accepting `long double`?

File: cpp/extensions/cloud_optics/mo_cloud_optics.h [hunk 1]
```diff
@@ -256,7 +256,9 @@ class CloudOptics : public OpticalProps {
     using yakl::COLON;
     using yakl::intrinsics::size;
     using yakl::intrinsics::allocated;
-    using yakl::intrinsics::any;
+    #ifdef RRTMGP_EXPENSIVE_CHECKS
```
Comment (brhillman) [inline cpp/extensions/cloud_optics/mo_cloud_optics.h:259]: Why are the using statements protected in ifdefs now? Is there any harm in including them?

Comment (xyuan) [reply cpp/extensions/cloud_optics/mo_cloud_optics.h]: Yes, with the latest version of YAKL, there is no need to use the ifdef protection, I have restored to the previous version

**Review by brhillman — APPROVED** [review]

**Pull request merged**